cinema/views.py

```diff
@@ -21,24 +24,57 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
 
+    def get_permissions(self):
+        if self.action == "list" and self.request.user.is_authenticated:
+            self.permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+        if self.action == "destroy":
+            self.permission_classes = ()
```

Comment:
Setting `permission_classes` to an empty tuple in the `destroy` action of `GenreViewSet` effectively removes all permissions, which might not be the intended behavior. Consider specifying appropriate permissions for this action.